Examples/Scripts/Python/MLAmbiguityResolution/ambiguity_solver_full_chain.py

```diff
@@ -0,0 +1,338 @@
+import glob
+import os
+import math
+
+import pandas as pd
+import numpy as np
+
+from sklearn.preprocessing import LabelEncoder, StandardScaler, OrdinalEncoder
+from sklearn.cluster import DBSCAN
+
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils
+
+import ast
+import onnxruntime as ort
+
+
+def PrepareDataSet(CKS_files: list[str]) -> pd.DataFrame:
+    """Read the dataset from the different file, remove the pure duplicate tracks and combine the datasets"""
+    """
+    @param[in] CKS_files: DataFrame contain the data from each track files (1 file per events usually)
+    @return: combined DataFrame containing all the track, ordered by events and then by truth particle ID in each event 
+    """
+    globalindex = 0
+    data = []
+    for f in CKS_files:
+        datafile = pd.read_csv(f)
+        # Remove tracks with less than 7 measurements
+        datafile = datafile[datafile["nMeasurements"] > 6]
+        datafile = datafile.sort_values("good/duplicate/fake", ascending=False)
+        # Remove pure duplicate (tracks purely identical) keep the ones good one if among them.
+        datafile = datafile.drop_duplicates(
+            subset=[
+                "particleId",
+                "Hits_ID",
+                "nOutliers",
+                "nHoles",
+                "nSharedHits",
+                "chi2",
+            ],
+            keep="first",
+        )
+        datafile = datafile.sort_values("particleId")
+        # Set truth particle ID as index
+        datafile = datafile.set_index("particleId")
+        # Transform the hit list from a string to an actual list
+        hitsIds = []
+        mergedIds = []
+        for list in datafile["Hits_ID"].values:
+            hitsIds.append(ast.literal_eval(list))
+        datafile["Hits_ID"] = hitsIds
+        # Combine dataset
+        data.append(datafile)
+    return data
+
+
+def PrepareData(data: pd.DataFrame) -> tuple[np.ndarray, np.ndarray]:
+    """Prepare the data"""
```

Comment:
what is the difference to `PrepareDataSet`?

Reply:
So prepare `PrepareDataSet` does the reading and formating of the csv input while `PrepareData` prepare the input of the NN (normalisation...). I am really bad at naming stuff very bad name for those function...

Reply:
The name of the function should be more expicit now.